tests/viewsets/test_locations.py

```diff
@@ -100,13 +100,16 @@ def test_create_with_bad_data(self):
         data_shapefile = dict(
             name='Arusha',
             radius=56.6789,
-            geopoint='POINT(30 10)',
-            shapefile=mocked_location_with_shapefile.shapefile)
+            geopoint='30,10',
+            shapefile=mocked_location_with_shapefile.shapefile,
+            )
 
         view2 = LocationViewSet.as_view({'post': 'create'})
         request2 = self.factory.post('/locations', data_shapefile)
         # Need authenticated user
         force_authenticate(request2, user=alice_user)
+        # make request mutable
+        request2.POST._mutable = True
```

Comment:
This seems too weird for me.  Let us find a way that does not need us to do this, OR give sufficient reasons why we cant find such a way.

My reasoning is that in the case of request.POST, it seems no activity in the server side should need to alter the request's data. And hence immutable objects are more suited, not to mention they have substantial performance advantages.

Reply:
Found an alternative way that doesn't require us setting the Post Mutability to True. Used it in the latest commit.

Simple changed the definitions of the data from data = dict() to data = {}